Doc/c-api/init.rst

```diff
@@ -919,8 +919,11 @@ from a C thread is::
 Note that the ``PyGILState_*`` functions assume there is only one global
 interpreter (created automatically by :c:func:`Py_Initialize`).  Python
 supports the creation of additional interpreters (using
-:c:func:`Py_NewInterpreter`), but mixing multiple interpreters and the
-``PyGILState_*`` API is unsupported.
+:c:func:`Py_NewInterpreter`), but switching between interpreters via the
+``PyGILState_*`` API is unsupported. With that being said, you still need
+to hold the :term:`GIL` in order to *create* a subinterpreter, even if using
+a per-interpreter GIL (see :pep:`684`). The need for :c:type:`PyGILState_STATE`
+doesn't go away when creating isolated subinterpreters!
```

Comment:
```suggestion
:c:func:`Py_NewInterpreter`), but switching between interpreters via the
``PyGILState_*`` API is unsupported. On the other hand, *creating*
sub-interpreters (whether they have a per-interprter GIL or not in
the sense of :pep:`684`) still require to hold the :term:`GIL`.
```

We are less enthusiastic in documentation in general. By the way, `PyGILState_STATE` is not documented but could be done in a separate PR.

Reply:
I think that's intentional. `PyGILState_Ensure` explicitly documents `PyGILState_STATE` as "opaque."

Reply:
Yes, but we can make a link to say that it's the public type of the opaque handle. (it could be a very short sentence but let's discuss it in a follow-up PR if needed)